contracts/HubPool.sol

```diff
@@ -51,9 +64,49 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
      *          LIQUIDITY PROVIDER FUNCTIONS         *
      *************************************************/
 
-    function addLiquidity(address token, uint256 amount) public {}
+    function addLiquidity(address l1Token, uint256 l1TokenAmount) public payable {
```

Comment:
Is this all new code? Or is it all copied? Or is it a mix? If a mix, can you comment on the parts that you have changed?

Reply:
it's copied but changed to include the new mapping structure. the v1 version of this did not specify the `l1Token` (one token per pool). the changes therefore are simply to define the l1Token within the mapping. the other small change is the mint logic is now called on the external contract, rather than on `this` as the LPtoken is external.